### ISSUE TITLE
Encountered upstream issues

### ISSUE THREAD

**Issue by archanox** [filipnavara/dotnet-riscv#1]
As the risc-v port is in progress by Samsung which doesn't appear to have a visible roadmap to the public, I don't think it's fair to start lodging tickets on the main repo.  Historically I've suggested making a "meta" post in the dotnet/dotnet repo, but it was closed by mods in favour of the WIP runtime issue, albeit being specifically just for the runtime repo and again, missing a channel to users for the visibility of the progress behind closed doors.

So consider this issue just a list of issues that may eventually get resolved once dotnet9 for risc-v is shipped. But until then, I believe it's good to track what users are encountering in the real world.

*** No AoT support ***
`/home/debian/.dotnet/sdk/9.0.100-preview.7.24407.1/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(90,5): error NETSDK1204: Ahead-of-time compilation is not supported on the current platform 'linux-riscv64'.`

**Comment by archanox** [comment]
Okay, I might have done a dumb. The projects need to targeting the dotnet 9 preview. Building projects may possibly unsupported forever?

**Comment by filipnavara** [comment]
NativeAOT support for RV64 is tracked upstream at https://github.com/dotnet/runtime/issues/106223.

The application host packages are built during the SDK build, but I am not exactly sure how they are supposed to be distributed for source builds. I think in the regular .NET scenarios they are available on NuGet but that's not really an option here.

**Comment by archanox** [comment]
> The application host packages are built during the SDK build, but I am not exactly sure how they are supposed to be distributed for source builds. I think in the regular .NET scenarios they are available on NuGet but that's not really an option here.

Yeah I think my understanding here is limited, but I'm guessing because I was trying to build a project that was targeting net8 and I was using your net9 preview build, it couldn't get the dependencies from the SDK, so either it failed there, or tried to download them from NuGet and then failed as you said. So I doubt targeting anything other than net9 will be supported. Which is fine by me.

**Comment by alitariq4589** [comment]
First of all, many thanks to @filipnavara and whoever was involved in releasing this dotnet release.

I am assuming this is the issue where we are tracking the issues faced by the user (correct me if I am wrong). 

My issue is related to running [GitHub Runner](https://github.com/actions/runner) on RISC-V. Upstream maintainers are not accepting the PRs/Changes for RISC-V (for the same reason as the .NET release of RISC-V). So I am maintaining my own Fork of [GitHub Runner for RISC-V](https://github.com/alitariq4589/github-runner-riscv). 

Github runner source code supports building it with version 8.0.417. But it is not available for RISC-V, so I modified the source code of Github runner to build with .NET version 10.0.102, which is released by you (the workflow with the changes is [here](https://github.com/alitariq4589/github-runner-riscv/actions/runs/22094847425/workflow), the release is [here](https://github.com/alitariq4589/github-runner-riscv/releases)). This is being used in the production environment.

I checked the patch for dotnet SDK version 10.0.102 in the workflow, and according to what I understand, the source code is only patched for musl and not for glibc. My understanding is that dotnet is not patched for glibc, and it comes as is from the upstream (correct me if I am wrong).


The issue I am facing is that, after some time inside the build, the runner simply fails without any error statement on the GitHub UI (typically, it is in annotations). 

I am attaching the `_diag log` of the job:

```
[2026-02-26 20:39:58Z INFO JobServerQueue] Got a step log file to send to results service.
[2026-02-26 20:39:58Z INFO JobServerQueue] Starting upload of step log file to results service ResultsLog, /home/runner/_diag/blocks/a45f706f-feef-4622-8030-a7d771ca64d8_c5c3cb4e-5537-49e9-96df-0376a2f573da.1
[2026-02-26 20:39:58Z INFO AddMaskCommandExtension] Add new secret mask with length of 76
[2026-02-26 20:39:58Z INFO JobServerQueue] Try to append 1 batches web console lines for record 'd7d8156a-a1d0-4533-8c81-c8f8515809ba', success rate: 1/1.
[2026-02-26 20:39:59Z INFO JobServerQueue] Tried to upload 2 file(s) to results, success rate: 2/2.
[2026-02-26 20:40:00Z INFO ProcessInvokerWrapper] STDOUT/STDERR stream read finished.
[2026-02-26 20:40:00Z INFO ProcessInvokerWrapper] STDOUT/STDERR stream read finished.
[2026-02-26 20:40:00Z INFO ProcessInvokerWrapper] Finished process 1385352 with exit code 139, and elapsed time 00:00:02.1811843.
[2026-02-26 20:40:00Z INFO CreateStepSummaryCommand] Step Summary file (/home/runner/_work/_temp/_runner_file_commands/step_summary_151827d5-ebce-43b3-8ffe-2bbfcc1db45e) is empty; skipping attachment upload
[2026-02-26 20:40:00Z INFO StepsRunner] Step result: Failed
[2026-02-26 20:40:00Z INFO ExecutionContext] Publish step telemetry for current step {
  "action": "actions/checkout",
  "ref": "v6",
  "type": "node24",
  "stage": "Main",
  "stepId": "d7d8156a-a1d0-4533-8c81-c8f8515809ba",
  "stepContextName": "checkout",
  "hasPreStep": false,
  "hasPostStep": true,
  "result": "failed",
  "errorMessages": [],
  "executionTimeInSeconds": 3,
  "startTime": "2026-02-26T20:39:57.6004461Z",
  "finishTime": "2026-02-26T20:40:00.0166919Z"
}.
```

This is a segmentation fault (error code 139). I faced this segmentation fault with previous versions of .NET pretty frequently in `qemu-system-riscv64`, which is why I am thinking that it is related to .NET instead of github runner source code. Additionally, it is pretty random and has occurred in clone and post-clone in different jobs (which is an indication that it is something related to .NET imo)

One other thing is that I am using this runner in another image, which is riscv64/ubuntu in the dockerhub. That is not an actual image from Canonical (that is because when I created the container image of github runner, there was no official Ubuntu image, and the production environment is still using the older one). I will try to change the image from my side to see if that resolves the issue, but I will really appreciate your insights on this.
